solver/src/settlement_submission/retry.rs

```diff
@@ -0,0 +1,136 @@
+use super::{EncodedSettlement, GAS_PRICE_REFRESH_INTERVAL, MAX_GAS};
```

Comment:
It feels to me as if this module is collecting a bunch of parts that are not really related to one another. Maybe the gas_price_stream should be part of the `gas_price_estimation` module and SettleResult/SettlementSender also seem not strictly related to retry. Maybe this module could be called transaction?

Reply:
Moved the stream to it's own module. Don't like it in gas_price_estimation because that is just responsible for creating the estimator whereas the stream is only used for retry.

Reply:
> SettleResult/SettlementSender also seem not strictly related to retry. Maybe this module could be called transaction

Imo they are because these types are only used for retry.